### PR TITLE
Set delegated access in session instead of url

### PR DESCRIPTION
**Pull request by kevin-verschaeve**

### CODE REVIEW

File: src/Controller/EnMarche/DelegatedAccessController.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+class DelegatedAccessController extends Controller
+{
+    use CanaryControllerTrait;
+
+    /**
+     * @Route("/espace-partage/{uuid}", name="access_delegation_set")
```
Comment (ottaviano) [inline src/Controller/EnMarche/DelegatedAccessController.php:17]: could you prefix by `app_` the route name + add `Method={GET}`, please?

File: src/Controller/EnMarche/DelegatedAccessController.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     */
+    public function delegatedSpace(DelegatedAccess $delegatedAccess, SessionInterface $session)
+    {
+        $this->disableInProduction();
+
+        $session->set(DelegatedAccess::ATTRIBUTE_KEY, $delegatedAccess->getUuid()->toString());
+
+        $routes = DelegatedAccessEnum::getFirstRoutesForType($delegatedAccess->getType());
+
+        return $this->redirectToRoute($routes[$delegatedAccess->getAccesses()[0]]);
```
Comment (ottaviano) [inline src/Controller/EnMarche/DelegatedAccessController.php:27]: how do you handle the empty `accesses` value? `getAccesses()[0]` will throw an exception?

File: src/Controller/EnMarche/DelegatedAccessController.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    /**
+     * @Route("/espace-standard/{type}", name="access_delegation_unset")
```
Comment (ottaviano) [inline src/Controller/EnMarche/DelegatedAccessController.php:31]: same for route prefix?

File: src/Entity/MyTeam/DelegatedAccessEnum.php [hunk 4]
```diff
@@ -34,4 +34,17 @@ public static function getAccessesForType(string $type): array
...
+
+    public static function getFirstRoutesForType(string $type): array
+    {
+        return [
+            DelegatedAccess::ACCESS_MESSAGES => "app_message_{$type}_list",
+            DelegatedAccess::ACCESS_EVENTS => "app_{$type}_event_manager_events",
+            DelegatedAccess::ACCESS_ADHERENTS => "app_{$type}_managed_users_list",
+            DelegatedAccess::ACCESS_COMMITTEE => "app_{$type}_committee",
+            DelegatedAccess::ACCESS_CITIZEN_PROJECTS => "app_{$type}_citizen_projects_list",
+            DelegatedAccess::ACCESS_JECOUTE => "app_jecoute_{$type}_list",
```
Comment (ottaviano) [inline src/Entity/MyTeam/DelegatedAccessEnum.php:46]: `_delegated_local_surveys_list`

File: src/Twig/AdherentDelegatedAccessesExtension.php [hunk 5]
```diff
@@ -50,56 +44,10 @@ public function getCurrentUser(Request $request): UserInterface
         return $user;
     }
 
-    public function getDelegatedAccess(Request $request)
+    public function getDelegatedAccess()
```
Comment (ottaviano) [inline src/Twig/AdherentDelegatedAccessesExtension.php:47]: typehint missing

File: src/Controller/EnMarche/DelegatedAccessController.php (hunk 2 above)
Comment (kevin-verschaeve) [reply src/Controller/EnMarche/DelegatedAccessController.php:27]: this controller is not and must not be called when the delegated accesses does not have any accesses in it. But, I can add a check and throw an exception if this is the case

**Pull request merged**